[coderhuang559] iP

coderhuang559 opened the pull request:
### DukePro

> “Your mind is for having ideas, not holding them.” – David Allen

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://nusmods.com/timetable/sem-1)
2. double-click it
3. add your tasks.
4. let it manage tasks for you 😊

And it is **FREE**
Features:

- [X] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon)
    


If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
     public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

coderhuang559 committed:
replace duke start msg

coderhuang559 committed:
add echo function

coderhuang559 committed:
Added add, list

coderhuang559 committed:
Create Task class

coderhuang559 committed:
Update array type to Task

coderhuang559 committed:
Update Task class

coderhuang559 committed:
Add mark as done feature

coderhuang559 committed:
Create todo, event, deadline class

coderhuang559 committed:
Add ToDo, Event, Deadline feature

coderhuang559 committed:
Add automated text UI testing

coderhuang559 committed:
Handle empty todo, invalid input

coderhuang559 committed:
Handle emptydate, emptytask, invalidinput exceptions

coderhuang559 committed:
Handle outofrange exception

coderhuang559 committed:
Compartmentalise main into methods

coderhuang559 committed:
Add documentation

coderhuang559 committed:
Add delete feature

coderhuang559 committed:
Update testcase

coderhuang559 committed:
Fix recurring ArrayList usage

ArrayList declaration is moved to Duke class from main. Can now be accessed by methods without being passed as param.

coderhuang559 committed:
Add func to write into tasks.txt

Duke.java/main writes list into ip/tasks.txt when user types "bye".

coderhuang559 committed:
Fix toString() in tasks

Print format into tasks.txt changed for easier delimeter removal when reading from file.

coderhuang559 committed:
Save task list at "bye"

coderhuang559 committed:
Save task list every change

Task list is updated after every mark/unmark/add/delete.

coderhuang559 committed:
Merge branch 'branch-Level-7'

coderhuang559 committed:
Delete repeated code

Delete repeated code in "bye".

coderhuang559 committed:
Add date

Dates are now identified as dates, not Strings.

coderhuang559 committed:
Add time

Date + Time inputs will be recognised as DateTime objects. Date inputs will also be recognised as DateTime objects with time = 0000.

coderhuang559 committed:
Clean up

Delete unused import.

coderhuang559 committed:
Merge branch 'branch-Level-8'

coderhuang559 committed:
Add classes for OOP

Extract methods from Duke to create other methods. Yet to extract content for Parser class.

coderhuang559 committed:
Add run() to Duke class

Yet to extract if-else chunk from run() into Parse class.

coderhuang559 committed:
Group methods into classes

Each class contains related methods.

coderhuang559 committed:
Merge branch 'branch-A-MoreOOP'

coderhuang559 committed:
Create packages

Create duke and exceptions packages. Exception classes are stored in package exceptions. The rest are stored in duke.

coderhuang559 committed:
Merge branch 'branch-A-Packages'

coderhuang559 committed:
Merge branch 'add-gradle-support'

coderhuang559 committed:
Add JUnit test for 2 methods

parseTaskFromString and markTask

coderhuang559 committed:
Merge branch 'branch-A-JUnit'

coderhuang559 committed:
Add Javadoc

Add Javadoc to all classes, except test classes.

coderhuang559 committed:
Add find feature

Give users a way to find a task by searching for a keyword.

coderhuang559 committed:
Merge branch 'branch-A-JavaDoc'

coderhuang559 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Deadline.java
#	src/main/java/duke/Task.java
#	src/main/java/duke/TaskList.java

coderhuang559 committed:
Update run()

Exception code chunks with similar functions are grouped together.

coderhuang559 commented:
### DukePro

> “Your mind is for having ideas, not holding them.” – David Allen

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://nusmods.com/timetable/sem-1)
2. double-click it
3. add your tasks.
4. let it manage tasks for you 😊

And it is **FREE**
Features:

- [X] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon)
    


If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
     public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

chonguschonguschongus commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,35 @@
+     * Prints a welcome message to the user.
+     */
+    public static void printWelcomeMessage() {
+        System.out.println("Hello! I'm BbabBBB");
```
What is a BbaBBB? 😆 🤣 🤖

coderhuang559 replied:
It's my **brother**'s name! I made this bot for him XD. Thanks for asking, `chonguschonguschongus`.

chonguschonguschongus commented on src/main/java/exceptions/DukeException.java:
```diff
@@ -0,0 +1,17 @@
+package exceptions;
```
Very good exception handling. I love it!

coderhuang559 replied:
`Thanks! `

chonguschonguschongus commented:
Lets Get To Merging this Pull Request

coderhuang559 committed:
GUI: Duke echoes User

Implemented GUI where Duke returns user input.

coderhuang559 committed:
GUI: Use incomplete Duke logic

Able to interact with Duke fully using GUI, but GUI is unable to print some messages.

HollaG commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,89 @@
+        } catch (FileUnloadableException e) {
+            System.out.println("File cannot be loaded.");
+        } catch (IOException e) {
+            System.out.println("e.getMessage");
```
I think you have extra quotes here!

HollaG commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,73 @@
+     *
+     * @return The formatted deadline in the format "d MMM yyyy h:mma".
+     */
+    public String getByFormatted() {
+        return this.by.format(DateTimeFormatter.ofPattern("d MMM yyyy h:mma"));
+    }
```
on first glance, I thought this function was 'getting something' by a formatted string. maybe change to `getFormattedDateBy`?

HollaG commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,157 @@
+     * @return The LocalDateTime object representing the date.
+     * @throws DateTimeParseException If the date string is in an invalid format.
+     */
+    public static LocalDateTime saveAsDate(String by) throws DateTimeParseException {
+        try {
+            // Try to parse the input as "yyyy-MM-dd" format
+            return LocalDateTime.parse(by, DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
+        } catch (DateTimeParseException e) {
+            by += " 0000";
+            // If parsing as "yyyy-MM-dd" format fails, try to parse as "yyyy-MM-dd HHmm" format
+            return LocalDateTime.parse(by, DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
+        }
+    }
```
I would suggest changing this to `parseDateFromString`? I think it's clearer cos it doesn't seem to be saving anything

HollaG commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,106 @@
+ */
+public class Task {
+    protected String taskName;
+    protected boolean status;
```
i would suggest using `isDone` bc status is ambigious i think!

HollaG reviewed (commented):
LGTM!

I really like your exception handling, it's very descriptive in what error you tell the user.
 
Your code was also easy to read (apart from that one bit about the getByFormatted). Good adherence to the style guide and formatting rules

coderhuang559 committed:
Implement GUI

GUI for users to interact with Duke.

coderhuang559 committed:
Merge branch 'branch-Level-10'

jrchoo commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,160 @@
+     * @return The LocalDateTime object representing the date.
+     * @throws DateTimeParseException If the date string is in an invalid format.
+     */
+    public static LocalDateTime saveAsDate(String by) throws DateTimeParseException {
+        try {
+            // Try to parse the input as "yyyy-MM-dd" format
+            return LocalDateTime.parse(by, DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
+        } catch (DateTimeParseException e) {
+            by += " 0000";
+            // If parsing as "yyyy-MM-dd" format fails, try to parse as "yyyy-MM-dd HHmm" format
+            return LocalDateTime.parse(by, DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
+        }
+    }
```
I agree with what HollaG has mentioned, saveAsDate doesn't really capture the essence of converting a string into another form.

jrchoo commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,160 @@
+     * @param taskArray The list of tasks to generate content for.
+     * @throws IOException If an I/O error occurs while writing to the file.
+     */
+    public static void generateTaskFileContent(ArrayList<Task> taskArray) throws IOException {
```
Method names are written as verbs, most people will be able to tell right away what the method does.

jrchoo commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,106 @@
+     * @return A string representation of the task's status and name.
+     */
+
+    public String statusAndTask() {
+        return "[Task]" + statusString() + " " + this.taskName;
+    }
```
Interesting to see that you have chosen statusAndTask as the name for this method for your Task class and its subclasses. Perhaps its a good idea to call this your toString() method instead?

jrchoo commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,106 @@
+ */
+public class Task {
+    protected String taskName;
+    protected boolean status;
```
I think it will better to name boolean values in the form of isDone to make it clearer.

jrchoo reviewed (commented):
LGTM.

coderhuang559 committed:
Add assertions

Use assert feature to document important assumptions that should hold at
various points in the code.

coderhuang559 committed:
Improve code quality

Improve readability of code, like shortening and compartmentalising
lengthy methods.

Improve comprehensibility of comments and code.

coderhuang559 committed:
Make sort function

Sort function allows users to sort tasks by date.

Sort command: User gets a sorted list of the tasks.
Tasks are sorted as such: ToDos, then according to dates.

Original sequence of tasks is preserved as tasks.txt remains the same.

coderhuang559 committed:
Merge pull request #3 from coderhuang559/branch-C-Sort

Make sort function

coderhuang559 committed:
Merge pull request #1 from coderhuang559/branch-A-Assertions

Add assertions

coderhuang559 committed:
Merge branch 'master' into branch-A-CodeQuality

coderhuang559 committed:
Merge pull request #2 from coderhuang559/branch-A-CodeQuality

Improve code quality

coderhuang559 committed:
Merge branch 'master' of https://github.com/coderhuang559/ip into branch-C-Sort

* 'master' of https://github.com/coderhuang559/ip:
  Improve code quality
  Add assertions

coderhuang559 committed:
Merge branch 'branch-C-Sort'

coderhuang559 committed:
Rename boolean

Rename boolean in Task class from status to isDone.

coderhuang559 committed:
Improve code quality

Extract methods in MainWindow.
Debug trimElements method in TaskList.
Extract methods in Parser.

coderhuang559 committed:
Upload GUI screenshot

Picture of Philip GUI when in use.

coderhuang559 committed:
Add UserGuide

Add UserGuide to README.

coderhuang559 committed:
Improve Javadoc

Modify Javadoc in all classes to adhere to coding style.

coderhuang559 committed:
Merge pull request #5 from coderhuang559/branch-A-UserGuide

Add UserGuide

coderhuang559 committed:
Fix layout issues

Increase width of label for better readability.

coderhuang559 committed:
Fix bugs

Fix non-functioning list command.
Fix non-functional jar file.

coderhuang559 committed:
Fix bugs

coderhuang559 committed:
Edit README

coderhuang559 committed:
Fix bugs

Recognise empty deadline and event descriptions.

Recognise empty from and to event dates.

coderhuang559 committed:
Fix bug

coderhuang559 committed:
Update build.gradle

Fix Linux launch problem
